### PR TITLE
Bug related to pinned task.

### PR DESCRIPTION
**Pull request by MonCaptain**
Unpinning the task from the main page would result in a "null style" error. This error results in preventing the user from making any future pinned tasks. This is now fixed.

### CODE REVIEW

File: react-app/pomodoro-timer-ui/src/Components/TodoTab/Todo.jsx [hunk 1]
```diff
@@ -38,23 +38,41 @@ export default function ({ todo, toggleComplete, removeTodo }) {
       setIsActivePin(false);
     }
   }
+
+
+  console.log("1", pinnedTodo.id)
```
Comment (Rebecca-CYLi) [inline react-app/pomodoro-timer-ui/src/Components/TodoTab/Todo.jsx]: found a console log

File: react-app/pomodoro-timer-ui/src/Components/TodoTab/Todo.jsx [hunk 2]
```diff
@@ -38,23 +38,41 @@ export default function ({ todo, toggleComplete, removeTodo }) {
...
   function handlePinButton() {
     // isActivePin is a boolean value that denotes if there's a todo pinned.
+    if (todo.is_completed){
+      console.log("todo was completed")
```
Comment (Rebecca-CYLi) [inline react-app/pomodoro-timer-ui/src/Components/TodoTab/Todo.jsx]: found a console log

File: react-app/pomodoro-timer-ui/src/Components/TodoTab/Todo.jsx [hunk 3]
```diff
@@ -38,23 +38,41 @@ export default function ({ todo, toggleComplete, removeTodo }) {
...
+      if (isActivePin) {
+        console.log(isActivePin)
```
Comment (Rebecca-CYLi) [inline react-app/pomodoro-timer-ui/src/Components/TodoTab/Todo.jsx]: found a console log

File: react-app/pomodoro-timer-ui/src/Components/TodoTab/Todo.jsx [hunk 4]
```diff
@@ -38,23 +38,41 @@ export default function ({ todo, toggleComplete, removeTodo }) {
...
+      document.getElementById(`${todo.id}`).style.fill = "white";
+      toggleComplete(todo)
+      setPinnedTodo(todo)
+      setIsActivePin(true)
+      return
+    }
 
     if (!isActivePin) {
       //when there's no pinned todo at the moment
+      console.log("no pin atm")
```
Comment (Rebecca-CYLi) [inline react-app/pomodoro-timer-ui/src/Components/TodoTab/Todo.jsx]: found a console log

File: react-app/pomodoro-timer-ui/src/Components/TodoTab/Todo.jsx [hunk 5]
```diff
@@ -38,23 +38,41 @@ export default function ({ todo, toggleComplete, removeTodo }) {
...
       setPinnedTodo(todo);
       document.getElementById(`${todo.id}`).style.fill = "white";
       setIsActivePin(true);
     } else if (todo.id != pinnedTodo.id && isActivePin) {
       /* when there's a pinned todo,
       but the user chooses to change the pin todo to something different */
+      console.log("todo clicked is different")
```
Comment (Rebecca-CYLi) [inline react-app/pomodoro-timer-ui/src/Components/TodoTab/Todo.jsx]: found a console log

File: react-app/pomodoro-timer-ui/src/Components/TodoTab/Todo.jsx [hunk 6]
```diff
@@ -38,23 +38,41 @@ export default function ({ todo, toggleComplete, removeTodo }) {
...
       document.getElementById(`${todo.id}`).style.fill = "white";
       document.getElementById(`${pinnedTodo.id}`).style.fill = "none";
       setPinnedTodo(todo);
     } else if (todo.id == pinnedTodo.id && isActivePin) {
       /* when the user chooses to unpin the current todo without replacing it 
       with a new pinned todo*/
+      console.log("todo clicked is the same")
```
Comment (Rebecca-CYLi) [inline react-app/pomodoro-timer-ui/src/Components/TodoTab/Todo.jsx]: found a console log

File: react-app/pomodoro-timer-ui/src/contexts/TodoContext.jsx [hunk 7]
```diff
@@ -57,17 +54,27 @@ export const TodoContextProvider = ({ children }) => {
   */
   const { authStates } = useAuthContext();
   function toggleComplete(todo) {
-    if (authStates.loggedIn){
-      apiClient.updateTask(todo)
+    if (authStates.loggedIn) {
+      apiClient.updateTask(todo);
     }
+    if (todo.id == pinnedTodo.id) {
+      console.log("todo completed was also pinned")
```
Comment (Rebecca-CYLi) [inline react-app/pomodoro-timer-ui/src/contexts/TodoContext.jsx]: found a console log

**Review by Rebecca-CYLi — CHANGES_REQUESTED** [review]
found console logs, please fix them

**Review by MonCaptain — COMMENTED** [review]
I forgot that with debugging, comes lots of console.logs. Should be good now

**Review by Rebecca-CYLi — APPROVED** [review]
revisions made, LGTM

**Pull request merged**